evalml/pipelines/components/estimators/classifiers/xgboost_classifier.py

```diff
@@ -88,11 +94,18 @@ def _convert_bool_to_int(X):
             col: "Integer" for col in X.ww.select("boolean", return_schema=True).columns
         }
 
+    def _label_encode(self, y):
+        if not is_integer_dtype(y):
+            self._label_encoder = LabelEncoder()
+            y = pd.Series(self._label_encoder.fit_transform(y), dtype="int64")
+        return y
+
     def fit(self, X, y=None):
         X, y = super()._manage_woodwork(X, y)
         X.ww.set_types(self._convert_bool_to_int(X))
         self.input_feature_names = list(X.columns)
         X = _rename_column_names_to_numeric(X, flatten_tuples=False)
+        y = self._label_encode(y)
```

Comment:
I think we need to decode the predictions if we've encoded them?

Reply:
Good call!